zhaquirks/bosch/rbsh_trv0_zb_eu.py

```diff
@@ -0,0 +1,454 @@
+"""Device handler for Bosch RBSH-TRV0-ZB-EU thermostat."""
+
+from typing import Any
+
+from zigpy.quirks import CustomCluster
+from zigpy.quirks.v2 import add_to_registry_v2
+from zigpy.quirks.v2.homeassistant.number import NumberDeviceClass
+import zigpy.types as t
+from zigpy.zcl.clusters.hvac import (
+    ControlSequenceOfOperation,
+    Thermostat,
+    UserInterface,
+)
+from zigpy.zcl.foundation import ZCLAttributeDef
+
+"""Bosch specific thermostat attribute ids."""
+
+# Mode of operation with values BoschOperatingMode.
+OPERATING_MODE_ATTR_ID = 0x4007
+
+# Valve position: 0% - 100%
+VALVE_POSITION_ATTR_ID = 0x4020
+
+# Remote measured temperature.
+REMOTE_TEMPERATURE_ATTR_ID = 0x4040
+
+# Window open switch (changes to a lower target temperature when on).
+WINDOW_OPEN_ATTR_ID = 0x4042
+
+# Boost preset mode.
+BOOST_ATTR_ID = 0x4043
+
+"""Bosch specific user interface attribute ids."""
+
+# Display orientation with values BoschDisplayOrientation.
+SCREEN_ORIENTATION_ATTR_ID = 0x400B
+
+# Displayed temperature with values BoschDisplayedTemperature.
+DISPLAY_MODE_ATTR_ID = 0x4039
+
+# Display on-time (5s - 30s).
+SCREEN_TIMEOUT_ATTR_ID = 0x403A
+
+# Display brightness (0 - 10).
+SCREEN_BRIGHTNESS_ATTR_ID = 0x403B
+
+# Control sequence of operation (heating/cooling)
+CTRL_SEQUENCE_OF_OPERATION_ID = 0x001B
+
+class BoschOperatingMode(t.enum8):
+    """Bosh operating mode attribute values."""
+
+    Schedule = 0x00
+    Manual = 0x01
+    Pause = 0x05
+
+
+class State(t.enum8):
+    """Binary attribute (window open) value."""
+
+    Off = 0x00
+    On = 0x01
+
+
+class BoschDisplayOrientation(t.enum8):
+    """Bosch display orientation attribute values."""
+
+    Normal = 0x00
+    Flipped = 0x01
+
+
+class BoschDisplayedTemperature(t.enum8):
+    """Bosch displayed temperature attribute values."""
+
+    Target = 0x00
+    Measured = 0x01
+
+
+class BoschControlSequenceOfOperation(t.enum8):
+    """Supported ControlSequenceOfOperation modes."""
+
+    Cooling = ControlSequenceOfOperation.Cooling_Only
+    Heating = ControlSequenceOfOperation.Heating_Only
+
+
+"""HA thermostat attribute that needs special handling in the Bosch thermostat entity."""
+SYSTEM_MODE_ATTR = Thermostat.AttributeDefs.system_mode
+
+"""Bosch operating mode to HA system mode mapping."""
+OPERATING_MODE_TO_SYSTEM_MODE_MAP = {
+    BoschOperatingMode.Schedule: Thermostat.SystemMode.Auto,
+    BoschOperatingMode.Manual: Thermostat.SystemMode.Heat,
+    BoschOperatingMode.Pause: Thermostat.SystemMode.Off,
+    "BoschOperatingMode.Schedule": Thermostat.SystemMode.Auto,
+    "BoschOperatingMode.Manual": Thermostat.SystemMode.Heat,
+    "BoschOperatingMode.Pause": Thermostat.SystemMode.Off,
```

Comment:
Are the strings really needed here? This is somewhat redundant.
(same for all other enum maps too)

Reply:
If my memory serves me right, when I was trying to write directly to the cluster the operating_mode "BoschOperatingMode.Manual", I was getting a string in the write_attributes.

Reply:
Removed.